test/volume_test.cpp

```diff
@@ -57,40 +60,92 @@ void test_volume(FilePath f, NT expected, NT tolerance=0.1)
     CHECK(error < tolerance);
 }
 
+template <typename NT, class string>
+void call_tests(string test) {
+
+    string strcube("cube");
+    string strcross("cross");
+    string strbirk("birk");
+    string strprod_simplex("prod_simplex");
+    string strsimplex("simplex");
+    string strskinny_cube("skinny_cube");
+
+    if (test.compare(strcube) == 0) {
+        test_volume<NT>("../data/cube10.ine", 1024.0);
+        test_volume<NT>("../data/cube20.ine", 1048576.0);
+        test_volume<NT>("../data/cube30.ine", 1073742000.0, 0.2);
+    }
+
+    if (test.compare(strcross) == 0) {
+        test_volume<NT>("../data/cross_10.ine", 0.0002821869);
+    }
+
+    if (test.compare(strbirk) == 0) {
+        test_volume<NT>("../data/birk3.ine", 0.125);
+        test_volume<NT>("../data/birk4.ine", 0.000970018);
+        test_volume<NT>("../data/birk5.ine", 0.000000225);
+        test_volume<NT>("../data/birk6.ine", 0.0000000000009455459196, 0.5);
+    }
+
+    if (test.compare(strprod_simplex) == 0) {
+        test_volume<NT>("../data/prod_simplex_5_5.ine", std::pow(1.0 / factorial(5.0), 2));
+        test_volume<NT>("../data/prod_simplex_10_10.ine", std::pow(1.0 / factorial(10.0), 2));
+        test_volume<NT>("../data/prod_simplex_15_15.ine", std::pow(1.0 / factorial(15.0), 2));
+        test_volume<NT>("../data/prod_simplex_20_20.ine", std::pow(1.0 / factorial(20.0), 2));
+    }
+
+    if (test.compare(strsimplex) == 0) {
+        test_volume<NT>("../data/simplex10.ine", 1.0 / factorial(10.0));
+        test_volume<NT>("../data/simplex20.ine", 1.0 / factorial(20.0));
+        test_volume<NT>("../data/simplex30.ine", 1.0 / factorial(30.0));
+        test_volume<NT>("../data/simplex40.ine", 1.0 / factorial(40.0));
+        test_volume<NT>("../data/simplex50.ine", 1.0 / factorial(50.0));
+    }
+
+    if (test.compare(strskinny_cube) == 0) {
+        test_volume<NT>("../data/skinny_cube10.ine", 102400.0);
+        test_volume<NT>("../data/skinny_cube20.ine", 104857600.0);
+    }
+}
 
 TEST_CASE("cube") {
-    test_volume("../data/cube10.ine", 1024);
-    test_volume("../data/cube20.ine", 1048576);
-    test_volume("../data/cube30.ine", 1073742000, 0.2);
+    std::string strtest("cube");
```

Comment:
Why not just create functions `call_test_cube<NT>()`, `call_test_cross<NT>()`, etc. and call then from here. This avoids all `if` conditions above and the use of all strings.